### PR TITLE
Rename Dropdown to Select

### LINKED ISSUE #444: Dropdown/Select v1.0.0
## Make dropdown component ready for publishing of v1.0.0

The following should be ready in the style (`jkl-dropdown`) and react (`jkl-dropdown-react`) packages before we can publish v1.0.0

- [ ] Package should be renamed to `jkl-select`/`jkl-select-react`
- [ ] Components should be renamed to `Select` and `NativeSelect`

### Style package should…
- [x] …have a `README.md` file containing _something_ that is **not** "TODO: …", preferrably simple usage instructions
- [x] …use only variables from `jkl-core` to define spacing, colors and font size/line height
- [x] …cover all variants found in the design sketches

### React package should…
- [x] …have a `README.md` file containing link to documentation on the portal
- [ ] …properly document all types/props in use and when/how to use them
- [x] …implement all variants present in style package
- [x] …implement all expected functionality (see below)
- [x] …generate WCAG-compliant and accessible HTML

### Expected functionality
- [x] Can have a preselected value
- [x] Accepts both strings and objects as items
- [x] Accepts help- and error messages
- [x] Accepts a placeholder
- [x] Accepts an `onChange` function
- [x] Accepts custom `className`s (e.g. for spacing)
- [x] Can be forced to be small even on large screens

### PR DESCRIPTION
**Pull request by piofinn**
affects: @fremtind/jkl-datepicker-react, @fremtind/jkl-datepicker, @fremtind/jkl-select-react,
@fremtind/jkl-select

## 📥 Proposed changes

Our dropdown was not a dropdown. This removes (deprecates) the `dropdown` packages and replaces them with packages named `select`. The components are renamed from `Dropdown` and `Select` to `Select` and `NativeSelect` respectively.

This also adds developer documentation to the react component, meaning this should close #444 

## ☑️ Submission checklist

-   [x] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [x] `yarn build` works locally with my changes
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] I have added necessary documentation (if appropriate)

## 💬 Further comments

This will break a lot of stuff, so should probably not be merged until we're ready to release v1

### CODE REVIEW

File: packages/select-react/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# [`@fremtind/jkl-select-react`](https://fremtind.github.io/jokul/components/dropdown/)
+
+## Beskrivelse
+
+[Bruk og prinsipper](https://fremtind.github.io/jokul/components/dropdown/) er beskrevet i `@fremtind/jkl-select` stil-pakken.
```
Comment (lfbergee) [inline packages/select-react/README.md]: Lenka til portalen er fortsatt dropdown

File: packages/select-react/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# [`@fremtind/jkl-select-react`](https://fremtind.github.io/jokul/components/dropdown/)
```
Comment (lfbergee) [inline packages/select-react/README.md]: Samme her

File: packages/select-react/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+2. Importer _både_ React-pakken (velg komponenten(e) du trenger) og stil-pakken i prosjektet ditt:
+
+```js
+import { Select, NativeSelect } from "@fremtind/jkl-select-react";
+import "@fremtind/jkl-select/select.min.css";
+```
+
+### Bruk
+
+De to versjonene av select benytter seg av forskjellige tilnærminger: `Select` konstruerer en nedtrekksliste ved hjelp av JavaScript (via pakken `@nrk/core-toggle`) slik at vi kan gi den stil som matcher den visuelle profilen vår. `NativeSelect` bruker et vanlig `<select>`-element fra HTML.
```
Comment (lfbergee) [inline packages/select-react/README.md]: Kanskje si noe om at NativeSelect kan være å foretrekke på lange lister i mobil feks? Fordi man da får brukt OSet sin innebygde liste-velger

File: packages/select/README.md [hunk 4]
```diff
@@ -1,8 +1,10 @@
-# [`@fremtind/jkl-dropdown`](https://fremtind.github.io/jokul/components/dropdown/)
+# [`@fremtind/jkl-select`](https://fremtind.github.io/jokul/components/dropdown/)
```
Comment (lfbergee) [inline packages/select/README.md]: Dropdown

File: portal/src/pages/components/index.tsx [hunk 5]
```diff
@@ -50,7 +50,7 @@ export default function AllComponents() {
             <Section title="Radioknapper" link="radiobutton" example={<RadioButtonExample />} />
             <Section title="Avmerkingsbokser" link="checkbox" example={<CheckboxExample />} />
             <Section title="Toggle-knapp" link="toggleswitch" example={<ToggleSwitchExample />} />
-            <Section title="Nedtrekksliste" link="dropdown" example={<DropdownExample />} />
+            <Section title="Nedtrekksliste" link="dropdown" example={<SelectExample />} />
```
Comment (lfbergee) [inline portal/src/pages/components/index.tsx]: dropdown

**Comment by piofinn** [comment]
Har fiksa opp linkene som fortsatt het "dropdown", og har laget en PR #484  med dokumentasjon om at `jkl-dropdown` utgår.

File: packages/datepicker/datepicker.scss [hunk 6]
```diff
@@ -109,7 +109,7 @@ $bottom-margin: $component-spacing--large;
         position: relative;
         flex-grow: 2;
 
-        & > .jkl-dropdown__value {
+        & > .jkl-select__value {
```
Comment (Saegrov) [inline packages/datepicker/datepicker.scss]: I think we should create our own css class and not reuse the jkl-select__value.

Comment (piofinn) [reply packages/datepicker/datepicker.scss]: Good catch!

File: packages/datepicker/datepicker.scss [hunk 7]
```diff
@@ -103,13 +103,17 @@ $bottom-margin: $component-spacing--large;
     &__year-selector {
         max-width: 30%;
         margin-right: $component-spacing--xxl;
+
+        & > .jkl-datepicker__year-value {
```
Comment (Saegrov) [inline packages/datepicker/datepicker.scss]: Do you need this as a nested selector? Why not just put it as a normal BEM-class?

File: packages/datepicker/datepicker.scss [hunk 8]
```diff
@@ -103,13 +103,17 @@ $bottom-margin: $component-spacing--large;
...
+            text-align: left;
+        }
     }
 
     &__month-selector {
         position: relative;
         flex-grow: 2;
 
-        & > .jkl-dropdown__value {
+        & > .jkl-datepicker__month-value {
```
Comment (Saegrov) [inline packages/datepicker/datepicker.scss]: Same as above

File: packages/select-react/package.json [hunk 9]
```diff
@@ -1,13 +1,14 @@
...
+    "name": "@fremtind/jkl-select-react",
+    "version": "0.0.0",
     "publishConfig": {
         "access": "public"
     },
-    "description": "Jøkul react dropdown component",
+    "description": "Jøkul react select component",
     "homepage": "https://fremtind.github.io/jokul",
     "keywords": [
         "dropdown",
```
Comment (Saegrov) [inline packages/select-react/package.json]: Don't know if this affects anything but remove dropdown keyword?

Comment (piofinn) [reply packages/select-react/package.json]: Replace all is _very_ effective 😅 Should probably add back "dropdown" as a keyword, yes.

Comment (piofinn) [reply packages/select-react/package.json]: I think this is just for searching on npmjs and so on. I'm thinking that it should stay, maybe? I don't think we're the only ones mixing up these two components 😅

**Pull request merged**